src/util/triplea/MapXMLCreator/ImageScrollPanePanel.java

```diff
@@ -0,0 +1,309 @@
+					public void run()
+					{
+						
+						m_imagePanel.scrollRectToVisible(new Rectangle(vp, vport.getSize()));
+					}
+				});
+				pp.setLocation(cp);
+			}
+			
+			public void mousePressed(MouseEvent e)
+			{
+				m_imagePanel.setCursor(hndCursor);
+				pp.setLocation(e.getPoint());
+			}
+			
+			public void mouseReleased(MouseEvent e)
+			{
+				m_imagePanel.setCursor(defCursor);
+				pp.setLocation(e.getPoint());
+			}
+		};
+		m_imagePanel.addMouseListener(imageMouseAdapter);
+		m_imagePanel.addMouseMotionListener(imageMouseAdapter);
+		
+		return m_imagePanel;
+	}
+	
+	private static void loadPolygons()
+	{
+		if (s_polygonsInvalid)
+		{
+			s_polygons.clear();
+			s_polygonsInvalid = false;
+		}
+		if (!s_polygons.isEmpty())
+			return;
+		File file = null;
+		if (MapXMLCreator.s_mapPolygonsFile == null)
+		{
+			if (MapXMLCreator.s_mapFolderLocation != null && MapXMLCreator.s_mapFolderLocation.exists())
+				file = new File(MapXMLCreator.s_mapFolderLocation, "polygons.txt");
+			if (file == null || !file.exists())
+				file = new File(MapXMLCreator.s_mapImageFile.getParent() + File.separator + "polygons.txt");
+		}
+		else
+		{
+			file = MapXMLCreator.s_mapPolygonsFile;
+		}
+		if (MapXMLCreator.s_mapPolygonsFile != null
+					|| file.exists()
+					&& JOptionPane.showConfirmDialog(new JPanel(), "A polygons.txt file was found in the map's folder, do you want to use the file to supply the territory shapes?", "File Suggestion",
+								1) == 0)
+		{
+			try
+			{
+				System.out.println("Load Polygons from " + file.getPath());
+				s_polygons = PointFileReaderWriter.readOneToManyPolygons(new FileInputStream(file.getPath()));
+			} catch (final IOException ex1)
+			{
+				System.out.println("Something wrong with your Polygons file");
+				ex1.printStackTrace();
+			}
+		}
+		else
+		{
+			try
+			{
+				System.out.println("Select the Polygons file");
+				final String polyPath = new FileOpen("Select A Polygon File", MapXMLCreator.s_mapFolderLocation, ".txt").getPathString();
+				if (polyPath != null)
+				{
+					System.out.println("Polygons : " + polyPath);
+					s_polygons = PointFileReaderWriter.readOneToManyPolygons(new FileInputStream(polyPath));
+				}
+				else
+				{
+					System.out.println("Polygons file not given. Will run regardless");
+				}
+			} catch (final IOException ex1)
+			{
+				System.out.println("Something wrong with your Polygons file");
+				ex1.printStackTrace();
+			}
+		}
+		MapXMLCreator.s_mapPolygonsFile = file;
+	}
+
+	protected static String findTerritoryName(final Point p, final Map<String, List<Polygon>> m_polygons)
+	{
+		String seaName = "unknown";
+		// try to find a land territory.
+		// sea zones often surround a land territory
+		for (final String name : m_polygons.keySet())
+		{
+			final Collection<Polygon> polygons = m_polygons.get(name);
+			for (final Polygon poly : polygons)
+			{
+				if (poly.contains(p))
+				{
+					if (name.endsWith("Sea Zone") || name.startsWith("Sea Zone"))

```

Comment:
will do, my thinking was to have it as strict as possible as we are checking against territory names